### PR TITLE
Simplify casting in `tobytes` and other cleanup

### PR DESCRIPTION
**Pull request by jakirkham**
Improve code readability in `tobytes` by casting arguments and assigning them beforehand. Some other minor cleanup has been included for simplicity.

### CODE REVIEW

File: python/rmm/_lib/device_buffer.pyx [hunk 1]
```diff
@@ -72,12 +72,13 @@ cdef class DeviceBuffer:
         if s == 0:
             return b""
 
+        cdef cudaStream_t c_stream = <cudaStream_t>stream
```
Comment (kkraus14) [inline python/rmm/_lib/device_buffer.pyx]: IIRC this needs a double cast to handle using integers as the opaque handle for a stream in Python land.
```suggestion
        cdef cudaStream_t c_stream = <cudaStream_t><uintptr_t>stream
```

Comment (jakirkham) [reply python/rmm/_lib/device_buffer.pyx]: In this case `stream` is already typed as `uintptr_t`. So we should be good here 🙂

Comment (kkraus14) [reply python/rmm/_lib/device_buffer.pyx]: Github UI hid this from me 😄, looks good

**Comment by jakirkham** [comment]
Thanks Keith! 😄

**Pull request merged**